### DISCUSSION TITLE
Push a font up by a small number of pixels

### DISCUSSION THREAD

**Discussion by russ627** [olikraus/u8g2#2171] [Q&A]
Hi,

I am using **u8x8_font_inb46_4x8_n** on a 256x50 display  unfortunately the bottom part of the numbers are cut off, is there an easy way of moving the font up by a couple of pixels? Its easier to see the problem in the attached picture.

Thanks in advance!

![display](https://user-images.githubusercontent.com/54704710/236702212-574948b6-aa4a-43f3-9ed1-a76f84ce2787.jpg)

**Comment by olikraus** [comment]
Decrease the y position.

**Reply by russ627** [reply]
The  y position is set to 0, here is the code 


u8x8.drawString(0,0,time);

**Reply by olikraus** [reply]
oh, you are using u8x8, i missed this. With u8x8 you can not move pixel by pixel. What about using u8g2 then?

**Reply by russ627** [reply]
does it support the 1287 VFD display?, I thought it was just the u8x8 that did.

**Reply by russ627** [reply]
OK, many thanks I can see it does I will try and report back.

**Reply by olikraus** [reply]
> does it support the 1287 VFD display?, I thought it was just the u8x8 that did.

It should be there....

**Reply by russ627** [reply]
Hi again, I moved to a esp32 devkit 1 as nano does not have the space for the u8g2 libraries and the u8x8 works as expected however when I try to compile the u8g2 library I get this error message  

 ****_initializing argument 1 of 'U8G2_GP1287AI_256X50_1_4W_SW_SPI::U8G2_GP1287AI_256X50_1_4W_SW_SPI(const u8g2_cb_t*, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t)'
   public: U8G2_GP1287AI_256X50_1_4W_SW_SPI(const u8g2_cb_t *rotation, uint8_t clock, uint8_t data, uint8_t cs, uint8_t dc, uint8_t reset = U8X8_PIN_NONE) : U8G2() {
                                            ~~~~~~~~~~~~~~~~~^~~~~~~~

exit status 1

Compilation error: invalid conversion from 'uint8_t' {aka 'unsigned char'} to 'const u8g2_cb_t*' {aka 'const u8g2_cb_struct*'} [-fpermissiv_**e]**

I know that the esp32 uses a different compiler that is a lot more fussy about types and have managed to knife and fork simple programs so that the datatypes match but this is a bit complex for me. Is it a simple fix?

**Reply by olikraus** [reply]
Which version of u8g2 do you use? Did you use an official version for Arduino IDE? All in all, answers are difficult without knowing more about the code. At least for me u8g2 compiles without problem.

**Reply by russ627** [reply]
Hi, Its Arduino IDE 2.1.0 on Linux and the library is 2.33.15 from arduino library. The processor is ESP32 DEVKIT V1 If I compile it for this processor it gives errors but if I change the board to Mega 2560 it will compile without errors.

**Reply by olikraus** [reply]
I tried to install esp32 within my own Arduino IDE, but failed until now. I will create an issue for this. However I am not sure, when I will find time to test this for myself.

https://github.com/olikraus/u8g2/issues/2177

**Reply by olikraus** [reply]
I did some more tests. The above mentioned error does not appear on my test system.
I will close the issue.

**Comment by russ627** [comment]
Hi,

  It fails on this line in my system,, I have downloaded 2.33.8 which on the latest arduino update installs OK. 

U8G2_GP1287AI_256X50_1_4W_SW_SPI u8g2(/* clock=*/ 10, /* data=*/ 6, /* cs=*/ 8, /* dc=*/U8X8_PIN_NONE, /* reset=*/ 13);

In file included from /:   initializing argument 1 of 'U8G2_GP1287AI_256X50_1_4W_SW_SPI::U8G2_GP1287AI_256X50_1_4W_SW_SPI(const u8g2_cb_t*, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t)'
   public: U8G2_GP1287AI_256X50_1_4W_SW_SPI(const u8g2_cb_t *rotation, uint8_t clock, uint8_t data, uint8_t cs, uint8_t dc, uint8_t reset = U8X8_PIN_NONE) : U8G2() {
                                            ~~~~~~~~~~~~~~~~~^~~~~~~~

exit status 1

Compilation error: invalid conversion from 'int' to 'const u8g2_cb_t*' {aka 'const u8g2_cb_struct*'} [-fpermissive]

what are you using?

BR

Russ

**Comment by russ627** [comment]
OK got it, the arduino version does not have an example so the format is wrong and at the time the latest version on githum would not install but after an upgrade today does.
